packages/core/helpers/useDataVizClasses.ts

```diff
@@ -67,16 +67,20 @@ export default function useDataVizClasses(config, viewport = null) {
     if (legend?.style === 'gradient') ulClasses.push('patterns-only')
     return ulClasses
   }
-  const hasBorder = legend?.hideBorder ? 'no-border' : ''
+  const hasSideBorder = legend.position === 'side' && legend?.hideSideBorder ? 'no-side-border' : ''
+  const hasTopBottomBorder = legend?.hideTopBottomBorder && legend.position !== 'side' ? 'no-top-bottom-border' : ''
```

Comment:
Can we streamline the logic and user interactions? For instance, could we implement a single check for the legend borders? If the user selects the bottom legend, we’d show a specific border style, and if the legend is positioned on the side, we’d display a different border style without having to bloat the initial state file with different positions. 

Let me know if that makes sense or if its something custom thats needed here.